rbac/common/base/base_state.py

```diff
@@ -247,46 +260,221 @@ def _find_in_state_container(self, container, address, object_id, target_id=None
         )
         return None
 
-    def get_from_state(self, state, object_id, target_id=None):
-        """Gets an address from the blockchain state
-        (state object is available when message is handled by transaction processor)"""
+    def deserialize(self, address, data):  # pylint: disable=unused-argument
+        """Deserialize the data of a blockchain address"""
+        # pylint: disable=not-callable
+        container = self._state_container()
+        container.ParseFromString(data)
+        return container
+
+    def get_address(self, context, address):
+        """Get the deserialized data of a blockchain address"""
+        data = state_client.get_address(context=context, address=address)
+        if data:
+            return self.deserialize(address=address, data=data)
+        return None
+
+    def get_addresses(self, context, addresses):
+        """Get the list of blockchain addresses"""
+        return state_client.get_addresses(context=context, addresses=addresses)
+
+    def get_from_state_context(self, context, object_id, target_id=None):
+        """Gets an address from the blockchain state"""
         address = self.address(object_id=object_id, target_id=target_id)
+        container = self.get_address(context=context, address=address)
+        if container:
+            return self._find_in_state_container(
+                container=container,
+                address=address,
+                object_id=object_id,
+                target_id=target_id,
+            )
+        return None
+
+    def _get_new_state(self):
+        """Returns a new state container (protobuf) with a single store object"""
         # pylint: disable=not-callable
         container = self._state_container()
+        getattr(container, self._state_container_list_name).extend(
+            [self._state_object()]
+        )
+        store = getattr(container, self._state_container_list_name)[0]
+        return container, store
 
-        results = state_client.get_address(state=state, address=address)
-        if not list(results):
+    def get_from_input_state(self, inputs, input_state, object_id, target_id=None):
```

Comment:
I am having a hard time understanding all of the things this file is responsible for. 
The simple string manipulations (uppercasing, camel casing, etc) could be extracted into a dedicated case converter.
The validations could be extracted into a validator. 
The file is called base_state, but has getters for inbound and outbound state that are extracting content from messages. The getters themselves perform a series of checks and validations, but in some cases simply return None. I am assuming these are checks for available write space?

If this is a state manager, please name it as such. Maybe call it state_manager and subclasses transaction_state_manager and message_state_manager?